gamla/functional_generic.py

```diff
@@ -699,3 +699,26 @@ def count_by(f: Callable) -> Dict[Any, int]:
         compose_left(f, functional.wrap_tuple),
         lambda x, y: x + 1 if x else 1,
     )
+
+
+#: Like `stack` but doesn't require additional brackets.
+packstack = compose_left(functional.pack, stack)
+#: Runs `packstack` with given functions, then runs `all` on the output.
+allstack = compose_left(packstack, after(all))
+#: Runs `packstack` with given functions, then runs `any` on the output.
+anystack = compose_left(packstack, after(any))
+
+
+def prepare_and_apply(f):
```

Comment:
not sure about "prepare" here. maybe something like: "squash_higher_order_func" ?

in any case this looks like black magic :)

Reply:
a common exapmle i'm seeing now is look at a kg to understand what change you want to do, then actually do the change

Reply:
so you run the function once on the input to prepare it, then again to actually transform the value